src/api.rs

```diff
@@ -27,6 +27,36 @@ pub fn test() -> &'static str {
     "Hello, test!"
 }
 
+#[put("/package/<id>", data = "<package>")]
+pub async fn package_update(
+    id: String,
+    package: Json<Package>,
+    mod_db: &State<ModuleDB>,
+) -> (Status, &'static str) {
+    let mod_r = mod_db.read().await;
+    let res = mod_r.get(&id);
+    if res.is_none() {
+        return (Status::NotFound, "Package does not exist.");
+    }
+
+    let db = res.unwrap();
+
+    //update the package if metadata matches
+    if (package.metadata.Name == db.name)
+        & (package.metadata.Version == db.ver)
+        & (package.metadata.ID == db.id)
+    {
+        base64_to_zip(
+            package.data.Content.as_ref().unwrap().as_str(),
+            db.path.as_str(),
+        )
+        .await
+        .unwrap();
```

Comment:
Since the user might provide illegal base64, you'll have to handle the `Err` case.